packages/grafana-ui/src/types/panel.ts

```diff
@@ -30,6 +30,7 @@ export interface PanelProps<T = any> {
   width: number;
   height: number;
   replaceVariables: InterpolateFunction;
+  onChangeTimeRange: (timeRange: AbsoluteTimeRange) => void;
```

Comment:
@ryantxu adding this callback to PanelProps, any thoughts? only thing I am not 100% sure about is if we should support the text form as well (the string relative date math expressions), or we can wait and add that later as a second parameter. Or if we should use TimeRange (that has from:/to in DateTime, and raw with the string versions)